### ISSUE TITLE
WARNING: Secure coding is not enabled for restorable state!

### ISSUE THREAD

**Issue by mizlan** [neovide/neovide#2785]
**Describe the bug**
The warning appeares when I start Neovide using

```
$ /Applications/Neovide.app/Contents/MacOS/neovide
2024-09-09 09:32:51.588 neovide[57496:14648394] WARNING: Secure coding is not enabled for restorable state! Enable secure coding by implementing NSApplicationDelegate.applicationSupportsSecureRestorableState: and returning YES.
```

**To Reproduce**
1. run `/Applications/Neovide.app/Contents/MacOS/neovide` and notice the error

**Expected behavior**
no warning, because running the other `neovide` executable (the one not inside /Applications) does not cause this warning

**Desktop (please complete the following information):**
 - OS: macOS 14.1.1 Sonoma
 - Neovide Version 0.13.33
 - Neovim Version v0.11.0-dev-694+gc7379dc1f4

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**

https://clbin.com/tFvtF

**Comment by fboundp** [comment]
Beginning with macOS 12 Monterey, a new method of serializing window state was
introduced. The old method is vulnerable to process injection attacks. However,
due to compatibility concerns, the old method remained the default, and
applications had to opt in to the new method. They do so by responding
positively to the app delegate selector
`applicationSupportsSecureRestorableState:`.

https://developer.apple.com/documentation/appkit/nsapplicationdelegate/3762521-applicationsupportssecurerestora

I wondered why I could not reproduce this issue locally. Poking around, it appears
that the new method became the default when your application is built with a macOS SDK
version from macOS 14 or higher, and with a minimum deployment of macOS 14 or higher.

Neovide’s Cargo.toml has `osx_minimum_system_version = "10.11"`, but that
appears only to be used by cargo-bundle and not by rustc. (Also, cargo-bundle
is no longer used for creating the app bundle for releases.) As a result,
I would expect that builds of neovide with the recent SDK would not have this
warning.

I created a branch that implements `applicationSupportsSecureRestorableState:`
at <https://github.com/fboundp/neovide/tree/secure-restorable-state> if you want
to give it a try. It should eliminate the warning even with older SDKs and
deployment targets. But still, even if I build with `env
MACOSX_DEPLOYMENT_TARGET=11.0 cargo build` and run on macOS 14.1.1, I have not
been able to reproduce this issue, so I cannot be confident in the fix.

**Comment by fredizzimo** [comment]
We are building Neovide using macOS 12 (the oldest one supported by Github) https://github.com/neovide/neovide/blob/5483294f1b4644b9f43f30c14f9f67e43097d818/.github/workflows/build.yml#L40C36-L40C38.  

The reason why we are doing that, is that builds produced by never os versions are not compatible with older systems. And people are constantly asking us to support those, here for example https://github.com/neovide/neovide/issues/2679

**Comment by fboundp** [comment]
Makes sense. In case you’re not aware, the developer tools allow you to build using up-to-date OS, compiler, and SDK and still back-deploy to older releases. That’s what `MACOSX_DEPLOYMENT_TARGET` is about.

**Comment by fredizzimo** [comment]
Thanks, that sounds like something we should experiment with at some point.
